src/averageAges.js

```diff
@@ -54,11 +83,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  let childrenWithPresentMothers = people
+    .filter(child => people.some(({ name }) => name === child['mother']));
```

Comment:
```suggestion
    .filter(child => people.some(({ name }) => name === child.mother));
```
You can use syntaxis with dot as it's a simple property, the same in other places below